blockchain/blockchain.go

```diff
@@ -115,12 +117,12 @@ func (b *Blockchain) Height() (uint64, error) {
 	var height uint64
 	return height, b.database.View(func(txn db.Transaction) error {
 		var err error
-		height, err = chainHeight(txn)
+		height, err = ChainHeight(txn)
 		return err
 	})
 }
 
-func chainHeight(txn db.Transaction) (uint64, error) {
+func ChainHeight(txn db.Transaction) (uint64, error) {
```

Comment:
I don't think it is a good idea to make this function public because now the caller is expected to provide a transaction which can have unintended consequences on the state of the DB if we don't carefully think about who has access to the DB. 

When the blockchain struct was designed a conscious decision was made that the public API of the Blockchain would not require a transaction, which essentially meant that only the blockchain (and migration) was responsible for writing to the DB.

As far as I have seen this change is due to migration, I don't think migration necessitates making this function public, as migration code is temporary. Instead, we can copy-paste the code or look at other migrations to take inspiration. In general, migration requirements should have no impact on the design of the rest of the code base.

Reply:
After further investigation, it looks like this is the case with other functions which were made public for migration reasons (which I had missed earlier) too. I think those functions should also be made private (in a different PR though)

Reply:
Are you okay to leave this as public for now then?